src/v/kafka/server/group_manager.cc

```diff
@@ -359,17 +359,31 @@ static group_tx_cmd<T> parse_tx_batch(
     auto r = batch.copy_records();
     auto& record = *r.begin();
     auto key_buf = record.release_key();
-    kafka::request_reader buf_reader(std::move(key_buf));
-    auto tx_version = model::control_record_version(buf_reader.read_int16());
+
+    iobuf_parser buf_reader(std::move(key_buf));
+    auto tx_version = reflection::adl<int16_t>{}.from(buf_reader);
```

Comment:
you can get by with int8_t for version, and it might be convenient since we use int8_t for versions in serde v2.